### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by aukgit**

### CODE REVIEW

**Comment by aukgit** [comment]
@nafeem-evatix kindly follow git-flow in future https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow#:~:text=Gitflow%20Workflow%20is%20a%20Git,designed%20around%20the%20project%20release.

File: inmemory.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	"errors"
+
+	"github.com/google/uuid"
+)
+
+// InMemoryDB implements RepoHandler for a custom in memory database
+type InMemoryDB struct {
+	DB *map[string]interface{}
```
Comment (aukgit) [inline inmemory.go:11]: use lower case/unexport db as we are doing transaction using methods

`+`

use specific struct than a general one.

File: inmemory.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+func (in *InMemoryDB) Create(data interface{}) error {
+	uniqueID := uuid.New().String()
+	database := *in.DB
+	database[uniqueID] = data
+
+	return nil
+}
+
+// Get retrieves all data from the map
+func (in *InMemoryDB) Get() ([]interface{}, error) {
```
Comment (aukgit) [inline inmemory.go:24]: `in *InMemoryDB` not a good name.

options are (sorted from best to good)

- `inMemoryDB`
- `imd`
- `receiver` when keeping things general and possibility of interchange of method between data structures.

File: repohandler.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package repohandler
+
+//  Repohandler is a general interface
+//  for working with database
+
+// RepoHandler ...
+type RepoHandler interface {
```
Comment (aukgit) [inline repohandler.go:7]: Each handler should come from each interface

**Comment by aukgit** [comment]
@nafeem-evatix don't merge until discussion is finalized.

File: cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+	inmemory := repohandler.NewInMemoryDB()       // initializing in memory database
+	data := struct{ Name string }{Name: "Evatix"} // test data
+
+	// creating multiple datas
+	inmemory.Create(data)
+	inmemory.Create(data)
+	inmemory.Create(data)
+
+	fmt.Println(inmemory.Get())            // returns all data created earlier
```
Comment (aukgit) [inline cmd/main.go:20]: GetAll() / All()

File: inmemory.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+	delete(database, identifier)
+
+	return nil
+}
+
+// NewInMemoryDB returns pointer to an RepoHandler in this case
+// returns InMemoryDB struct because it has implemented RepoHandler
+func NewInMemoryDB() RepoHandler {
+	inMemoryDBMap := make(map[string]interface{})
```
Comment (aukgit) [inline inmemory.go:79]: use capacity/length when dealing with map or slice

map can only have a length

File: inmemory.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+	database[identifier] = data
+
+	return nil
+}
+
+// Delete removes data
+func (in *InMemoryDB) Delete(identifier string) error {
+	database := *in.DB
+	getValue := database[identifier]
```
Comment (aukgit) [inline inmemory.go:65]: _, has :=

File: inmemory.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+	if getValue == nil {
```
Comment (aukgit) [inline inmemory.go:67]: `if !has { ....`

File: repohandler.go (hunk 3 above)
Comment (nafeem-evatix) [reply repohandler.go:7]: Not quite sure what you meant bhaiya @aukgit

File: inmemory.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	delete(database, identifier)
+
+	return nil
+}
+
+// NewInMemoryDB returns pointer to an RepoHandler in this case
+// returns InMemoryDB struct because it has implemented RepoHandler
+func NewInMemoryDB() RepoHandler {
```
Comment (aukgit) [inline inmemory.go:78]: repohandler.RepoHandler, not good naming.

File: mongodb.go [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+// Delete ...
+// Todo : Implement Delete
+func (mdb *MongoDB) Delete(identifier string) error {
+	return nil
+}
+
+// NewMongoDB returns pointer to an RepoHandler in this case
+// returns MongoDB struct because it has implemented RepoHandler
+func NewMongoDB(dbSession *mgo.Session, DBName string, DBCollection string) RepoHandler {
```
Comment (aukgit) [inline mongodb.go:44]: mongodb.New() // could be

File: inmemory.go [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+	return nil
+}
+
+// Get retrieves all data from the map
+func (in *InMemoryDB) Get() ([]interface{}, error) {
+	database := *in.DB
+	var values []interface{}
+
+	for uniqueID := range database {
```
Comment (aukgit) [inline inmemory.go:28]: while looping make sure to loop from a plural item, database name is a plural form but expecting to have 

`for uniqueID := range uniqueIds {` something like this, which make sense.

**Review by aukgit — COMMENTED** [review]
feedbacks @nafeem-evatix

**Pull request merged**